src/engine/framework/System.cpp

```diff
@@ -51,6 +51,43 @@ SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 #include <sys/file.h>
 #endif
 
+#if defined(DAEMON_USE_FLOAT_EXCEPTIONS)
+	#if defined(__USE_GNU) || defined(__FreeBSD__)
+		#if defined(DAEMON_ARCH_amd64) || defined(DAEMON_ARCH_i686)
+			#include <cfenv>
+			#define DAEMON_USE_FLOAT_EXCEPTIONS_AVAILABLE
+		#endif
+	#elif defined (__APPLE__)
+		#if defined(DAEMON_ARCH_amd64) || defined(DAEMON_ARCH_arm64)
+			#include <cfenv>
+			#include <csignal>
+			#define DAEMON_USE_FLOAT_EXCEPTIONS_AVAILABLE
+		#endif
+	#elif defined(_WIN32)
+		#if defined(DAEMON_ARCH_amd64) || defined(DAEMON_ARCH_i686)
+			#include <cfloat>
+			#include <cfenv>
+			#define DAEMON_USE_FLOAT_EXCEPTIONS_AVAILABLE
+		#endif
+	#endif
+#endif
+
+#if defined(DAEMON_USE_FLOAT_EXCEPTIONS) && !defined(DAEMON_USE_FLOAT_EXCEPTIONS_AVAILABLE)
+	#warning Missing floating point exception implementation.
```

Comment:
This could be a `#else` in the preceding chain

Reply:
Done.